### PR TITLE
Pivotal ID # 177042374: Delete Draft After Submission

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/177042374

- Use interface in the resource in order to get the proper instance based on the persistence type
- Remove the any existing temporary draft after the submission is created

### CODE REVIEW

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/repositories/SubmissionMongoRepository.kt [hunk 1]
```diff
@@ -45,6 +45,8 @@ interface SubmissionDraftRepository : MongoRepository<DocSubmissionDraft, String
 
     fun deleteByUserIdAndKey(userId: String, key: String)
 
+    fun deleteByUserIdAndContentContaining(userId: String, content: String)
```
Comment (Juan-EBI) [inline submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/db/repositories/SubmissionMongoRepository.kt:48]: is not the accession the `key`? I just wondering if there is a better way to perform this operation as full document scan looks pretty heavy

File: submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt [hunk 2]
```diff
@@ -107,11 +109,36 @@ internal class SubmissionDraftApiTest(tempFolder: TemporaryFolder) : BaseIntegra
...
+                "accno" to "ABC-129"
+                "ReleaseDate" to "2021-09-21"
+                "type" to "Study"
+            }.toString()
+            val draftResponse = webClient.createSubmissionDraft(updatedDraft)
+
+            webClient.submitSingleFromDraft(draftResponse.key)
+
+            // TODO this approach must be improved once the testing for async submissions are in place
+            Thread.sleep(10000)
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt:139]: Yes, but we still wouldn't be testing the actual async feature which is the one the ST will be actually using. Yes, we won't have a TODO in the code but the test won't cover what it should actually test.

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt [hunk 3]
```diff
@@ -48,19 +48,19 @@ internal class SubmissionMongoPersistenceService(
...
 
         return submission
     }
 
-    private fun processDbSubmission(submission: ExtSubmission): ExtSubmission {
+    private fun processDbSubmission(submission: ExtSubmission, draftKey: String?): ExtSubmission {
         subDataRepository.expireActiveProcessedVersions(submission.accNo)
-        draftDocDataRepository.deleteByUserIdAndKey(submission.submitter, submission.accNo)
-        draftDocDataRepository.deleteByUserIdAndKey(submission.owner, submission.accNo)
+        deleteSubmissionDrafts(submission.submitter, submission.accNo, draftKey)
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt:62]: I don't think we're deleting twice, what the code is doing is:
- If a draft key is given, delete any draft that might exist with this key for both the submitter and the owner
- Delete any draft that might exist for the submission accession number for both the submitter and the owner

File: submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionDraftApiTest.kt:139]: ok

File: submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (Juan-EBI) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt:62]: we are calling delete the same key function ` draftKey?.let { draftDocDataRepository.deleteByUserIdAndKey(email, draftKey) }` two times one on each call `deleteSubmissionDrafts` I no?

[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionMongoPersistenceService.kt]: Right, I made the changes after validating with @nestor-diaz

**Review by Juan-EBI — APPROVED** [review]

**Pull request merged**